packages/core/src/dispatcher.ts

```diff
@@ -345,21 +345,34 @@ export class Dispatcher {
    */
   async _handleResponseMessage(message: ResponseMessage): Promise<void> {
     const { id: queryId, tips } = message
-    const expectedStreamID = this.messageBus.outstandingQueries.get(queryId)
-    if (expectedStreamID) {
-      const newTip = tips.get(expectedStreamID.toString())
-      if (!newTip) {
-        throw new Error(
-          "Response to query with ID '" +
-            queryId +
-            "' is missing expected new tip for StreamID '" +
-            expectedStreamID +
-            "'"
-        )
+    const outstandingQuery = this.messageBus.outstandingQueries.queryMap.get(queryId)
+    if(outstandingQuery == undefined){
+      console.warn(`Response to query with ID ${queryId} is not a valid outstanding query`)
```

Comment:
IMO, it is better to do
```
if (!outstandingQuery) {
  console.warn(`...`)
  return
}
// .. and here goes your else-block
```

Reply:
or, alternatively, there is a question mark operator in TS: https://www.delftstack.com/howto/typescript/question-mark-in-typescript/

Could work like `const expectedStreamID = outstandingQuery?.streamID`.